src/executables/service/locator/windows.ts

```diff
@@ -0,0 +1,102 @@
+import * as fs from 'fs-extra';
+import * as vscode from 'vscode';
+import path = require('path');
+import winreg = require('winreg');
+import { getUniquePaths, AbstractLocatorService } from './shared';
+
+export class WindowsExecLocator extends AbstractLocatorService {
+    constructor() {
+        super();
+        this.emitter = new vscode.EventEmitter<string[]>();
+        this._binaryPaths = [];
+    }
+    public refreshPaths(): void {
+        this._binaryPaths = getUniquePaths(Array.from(
+            new Set([
+                ...this.getHomeFromDirs(),
+                ...this.getHomeFromEnv(),
+                ...this.getHomeFromRegistry(),
+                // ... this.getHomeFromConda()
+            ])
+        ));
+        this.emitter.fire(this._binaryPaths);
+    }
+
+    private getHomeFromRegistry(): string[] {
+        const registryBins: string[] = [];
+        const potentialBins = [
+            new winreg({
+                hive: winreg.HKLM,
+                key: '\\SOFTWARE\\R-core\\R',
+            }),
+            new winreg({
+                hive: winreg.HKLM,
+                key: '\\SOFTWARE\\R-core\\R64',
+            })
+        ];
+
+        for (const bin of potentialBins) {
+            void new Promise(
+                (c, e) => {
+                    bin.get('InstallPath', (err, result) => err === null ? c(result) : e(err));
+                }
+            ).then((item: winreg.RegistryItem) => {
+                if (item) {
+                    const resolvedBin = item.value;
+                    const i386 = `${resolvedBin}\\i386\\`;
+                    const x64 = `${resolvedBin}\\x64\\`;
+
+                    if (fs.existsSync(i386)) {
+                        registryBins.push(i386);
+                    }
+
+                    if (fs.existsSync(x64)) {
+                        registryBins.push(x64);
+                    }
+                }
+            });
+        }
+
+        return registryBins;
```

Comment:
If I'm not mistaken, this returns an empty array that gets populated in an async manner later on.